pkg/internal/itest/testpeer/generator.go

```diff
@@ -195,6 +238,33 @@ func NewTestGraphsyncPeer(ctx context.Context, mn mocknet.Mocknet, p tnet.Identi
 	}
 
 	peer.DatatransferServer = dtRemote
+	peer.Graphsync = true
+	return peer, nil
+}
+
+func NewTestHttpPeer(ctx context.Context, mn mocknet.Mocknet, p tnet.Identity, t *testing.T) (TestPeer, error) {
+	peer, _, err := newTestPeer(ctx, mn, p)
+	if err != nil {
+		return TestPeer{}, err
+	}
+
+	// Create http multiaddr from random peer addr and add it to the peer's addreses
+	httpAddr := p.Address().Encapsulate(ma.StringCast("/http"))
+	peer.Host.Peerstore().AddAddr(p.ID(), httpAddr, 10*time.Minute) // TODO: Look into ttl duration?
```

Comment:
any idea what the impact of this is for testing? does it not work if we register it with the addressbook?

Reply:
Not sure what you mean by address book, but this seemed to be the only way I could add an address into the host's list of addresses. Thereisn't an exposed list of addresses anywhere.